models/repo/user_repo.go

```diff
@@ -166,3 +166,19 @@ func GetIssuePosters(ctx context.Context, repo *Repository, isPull bool) ([]*use
 	)
 	return users, db.GetEngine(ctx).Where(cond).OrderBy(user_model.GetOrderByName()).Find(&users)
 }
+
+// GetIssuePosters returns users with limit of 30 whose user name started with prefix that have authored an issue/pull request for the given repository
+// If isShowFullName is set to true, also include full name prefix search
+func GetIssuePostersWithPrefix(ctx context.Context, repo *Repository, isPull bool, prefix string, isShowFullName bool) ([]*user_model.User, error) {
+	users := make([]*user_model.User, 0, 8)
+	cond := builder.In("`user`.id",
+		builder.Select("poster_id").From("issue").Where(
+			builder.Eq{"repo_id": repo.ID}.
+				And(builder.Eq{"is_pull": isPull}),
+		).GroupBy("poster_id"),
+	)
+	if isShowFullName {
+		return users, db.GetEngine(ctx).Where(cond).Table("user").Where("name LIKE ? or full_name LIKE ?", prefix+"%", prefix+"%").OrderBy(user_model.GetOrderByName()).Limit(30).Find(&users)
```

Comment:
We can use `Select` to indicate the special columns to make it faster.

Reply:
Do you mean use [`builder.Select`](https://pkg.go.dev/xorm.io/builder@v0.3.12#Select) to directly select `id, name, fullname, avatar` from user table?